org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl

```diff
@@ -140,17 +152,26 @@ public class [javaClassNameForAdaptorServletListener(anAdaptorInterface) /] impl
     }
 
     private static String generateBasePath(final ServletContextEvent servletContextEvent) {
-        final ServletContext servletContext = servletContextEvent.getServletContext();
-        String base = getInitParameterOrDefault(servletContext, PROPERTY_BASE, DEFAULT_BASE);
-        return base + servletContext.getContextPath();
+        return (getBasePathFromEnvironment().orElseGet(() -> getBasePathFromContext(servletContextEvent).orElseGet(() -> DEFAULT_BASE))) +
+            servletContextEvent.getServletContext().getContextPath();
+    }
+
+    private static Optional<String> getBasePathFromEnvironment() {
+        final Map<String, String> env = System.getenv();
+        if(env.containsKey(BASE_PATH_ENV_KEY)) {
+            logger.info("Found {} env variable", BASE_PATH_ENV_KEY);
```

Comment:
If you are allowing people to customise the key, you should remove the field as I described above and pass it here as a parameter.

Reply:
See comment above.

Reply:
See the comment above :D

Reply:
fixed.